### ISSUE TITLE
May drop -l elements if specify both --libs and --cflags

### ISSUE THREAD

**Issue by ionenwks** [pkgconf/pkgconf#258]
Noticed with openssl and both pkgconf-1.9.0 and 1.9.1
```
$ pkgconf-1.9.1 --libs openssl
-lssl -lcrypto
$ pkgconf-1.9.1 --libs --cflags openssl
-lssl
$ pkgconf-1.8.0 --libs --cflags openssl
-lssl -lcrypto
```

openssl.pc:
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib64
includedir=${prefix}/include

Name: OpenSSL
Description: Secure Sockets Layer and cryptography libraries and tools
Version: 1.1.1q
Requires: libssl libcrypto
```

libssl.pc:
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib64
includedir=${prefix}/include

Name: OpenSSL-libssl
Description: Secure Sockets Layer and cryptography libraries
Version: 1.1.1q
Requires.private: libcrypto
Libs: -L${libdir} -lssl
Cflags: -I${includedir}
```

libcrypto.pc:
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib64
includedir=${prefix}/include
enginesdir=${libdir}/engines-1.1

Name: OpenSSL-libcrypto
Description: OpenSSL cryptography library
Version: 1.1.1q
Libs: -L${libdir} -lcrypto
Libs.private: -ldl -pthread 
Cflags: -I${includedir}
```

**Comment by kaniini** [comment]
It seems the new solver finds the `Requires: libcrypto` reference to be already processed due to the `Requires.private: libcrypto` in libssl.pc, because requires and private requires are tracked separately.  Solution is to merge the lists and have a dependency flag.

**Comment by kaniini** [comment]
(Of course the better solution is to not try to solve twice, once for cflags, and once for libs, separately.)

**Comment by kaniini** [comment]
Resolved in pkgconf 1.9.2.
